### ISSUE TITLE
Encountering NaNs when calculating IoU

### ISSUE THREAD

**Issue by Mayur28** [qubvel/segmentation_models.pytorch#761]
Hi,

For some reason, I am encountering NaNs when computing the IoU.
This is my code:
```
pred_mask, pred_label = model(val_input)
my_pred_label = pred_label.sigmoid() > 0.5
my_pred_mask = pred_mask.sigmoid() > 0.5
tp, fp, fn, tn = smp.metrics.get_stats(my_pred_mask.to(int), val_input["Mask"].to(int), mode='binary', threshold=0.5)
iou_score = smp.metrics.iou_score(tp, fp, fn, tn, reduction="micro")
```
The NaNs occur randomly for some reason. I have also ensured that `my_pred_mask.to(int)` and `val_input["Mask"].to(int)` do not contain NaNs using torch.isnan(my_pred_mask.to(int)) and torch.isnan(val_input["Mask"].to(int)). Furthermore, I have verified that the predicted mask and the groundtruth mask only contain {0, 1}, however, NaNs continue to occur occasionally.

Any assistance would be highly appreciated.

**Comment by qubvel** [comment]
Hi, were you able to identify the exact gt and pred masks that cause NaNs?
What are their shapes?
What are tp, fp, fn, and tn values?

**Comment by Mayur28** [comment]
Hi, I managed to resolve the issue. It was caused due to an error with my dataloader which was randomly returning blank groundtruth masks and because of this, the denominator of the IoU would be 0, thus, resulting in NaNs being produced. I have managed to resolve the issue, thanks!
